tests/integration/acp.go

```diff
@@ -112,7 +112,7 @@ func addPolicyACP(
 
 	nodeIDs, nodes := getNodesWithIDs(action.NodeID, s.nodes)
 	for index, node := range nodes {
-		ctx := getContextWithIdentity(s, action.Identity, nodeIDs[index])
+		ctx := getContextWithIdentity(s.ctx, s, action.Identity, nodeIDs[index])
```

Comment:
question: Curious why this was done?

Reply:
what exactly? We need a context with an identity in it for sending a request.
Please clarify your question. I also documented some of my decisions. Let me know, if they are not clear enough.

Reply:
because in some parts of the code the context is not stored in `s.ctx`. I don't know if there is a good reason for this.
I can actually change it to something like this:
```go
s.ctx = newCtx
updateContextWithIdentity(s, action.Identity, nodeIDs[index]) // will store in s.ctx
policyResult, err := node.AddPolicy(s.ctx, action.Policy)
```
Not sure if that makes it better, but I don't have a strong opinion here.

Reply:
> because in some parts of the code the context is not stored in `s.ctx`. I don't know if there is a good reason for this. I can actually change it to something like this:
> 
> ```go
> s.ctx = newCtx
> updateContextWithIdentity(s, action.Identity, nodeIDs[index]) // will store in s.ctx
> policyResult, err := node.AddPolicy(s.ctx, action.Policy)
> ```
> 
> Not sure if that makes it better, but I don't have a strong opinion here.

Oh wow, perhaps worth investigating later where and why that happens. I am happy with the suggested solution you gave for places that have missing ctx